python/cuda_parallel/cuda/parallel/experimental/iterators/_iterators.py

```diff
@@ -132,12 +137,24 @@ def advance(state, distance):
         raise NotImplementedError("Subclasses must override advance staticmethod")
 
     @staticmethod
-    def dereference(state):
+    def dereference(state, *args):
```

Comment:
Should this now be a `@property` rather than a `@staticmethod`? Ditto for `advance`.

Reply:
Should I make `advance` and `dereference` properties for all iterators, even those unaffected by iterator_io?

Reply:
Yes, I think we should